File: assets/src/book.py
# Book sample
# coding: utf-8

# Book 本
class Book():
	topic = ''
	def kind(self):
		return '本'
	def read(self):
		print(self.kind() + 'を読む')
	def setTopic(self, topic):
		self.topic = topic
	def getTopic(self):
		return self.topic
	def study(self):
		self.read()
		print(self.getTopic() + 'を学ぶ')

class Magazine(Book):
	def kind(self):
		return '雑誌'

book = Book()
book.setTopic('語学')
book.study()

book = Magazine()
book.setTopic('ビジネス')
book.study()

File: assets/src/book.cpp
/* Book Sample */

#include <iostream> // cout, endl を使うため
#include <string> // string を使うため

using namespace std;

/* Book: 本 */
class Book
{
private:
	string topic;

public:
	virtual string Kind()
	{
		return "本";
	}

	virtual void Read()
	{
		cout << this->Kind() << "を読む" << endl;
	}

	virtual void SetTopic(string topic)
	{
		this->topic = topic;
	}

	virtual string GetTopic()
	{
		return this->topic;
	}

	virtual void Study()
	{
		this->Read();
		cout << this->GetTopic() << "を学ぶ" << endl;
	}
};

class Magazine : public Book
{
	virtual string Kind()
	{
		return "雑誌";
	}
};

int main()
{
	Book* book1 = new Book();
	book1->SetTopic("語学");
	book1->Study();
	delete book1;

	Book* book2 = new Magazine();
	book2->SetTopic("ビジネス");
	book2->Study();
	delete book2;
}